projects/plugins/jetpack/class.jetpack.php

```diff
@@ -7652,4 +7652,32 @@ public static function get_products_for_purchase() {
 
 		return $products;
 	}
+
+	/**
+	 * Determine if the current user is allowed to make Jetpack purchases without
+	 * a WordPress.com account
+	 *
+	 * @return boolean True if the user can make purchases, false if not
+	 */
+	public static function current_user_can_purchase() {
+		$connection_status = \Automattic\Jetpack\Connection\REST_Connector::connection_status( false );
+
+		// The site must be connected to Jetpack.
+		if ( ! $connection_status['isActive'] ) {
+			return false;
+		}
+
+		// We cannot do userless purchases for sites that are not connected userlessly.
+		if ( $connection_status['hasConnectedOwner'] ) {
+			return false;
+		}
```

Comment:
@leogermani could speak to this some more, but would it make sense to use our helper methods in situations like this?
https://github.com/Automattic/jetpack/blob/51edd8fd3b366e620c82624792fa52922513b23f/projects/packages/connection/src/class-manager.php#L533-L541
https://github.com/Automattic/jetpack/blob/51edd8fd3b366e620c82624792fa52922513b23f/projects/packages/connection/src/class-manager.php#L582-L592

Reply:
Again, yes! `Connection\Manager` has all the methods you need to check the connection status.